internal/handlers.go

```diff
@@ -102,6 +105,183 @@ func PostAccount(w http.ResponseWriter, r *http.Request, params httprouter.Param
 		return
 	}
 
-	Accounts[account.AccountId] = account
+	models.Accounts[account.AccountId] = account
 	utils.WithPayload(w, http.StatusCreated, nil)
 }
+
+func GetAllTransactions(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+	transactions := []*models.Transaction{}
+
+	accountId := params.ByName("account-id")
+
+	_, found := models.Accounts[accountId]
+
+	if !found {
+		msg := "Account Not Found"
+		zerolog.TimeFieldFormat = zerolog.TimeFormatUnix
+		log.Info().Msg(msg)
+		utils.ErrorWithMessage(w, http.StatusNotFound, msg)
+		return
+	}
+
+	for _, transaction := range models.Transactions {
+		if transaction.Owner == accountId {
+			transactions = append(transactions, transaction)
+		}
+	}
+
+	res, err := jsoniter.Marshal(&transactions)
+
+	if err != nil {
+		zerolog.TimeFieldFormat = zerolog.TimeFormatUnix
+		log.Error().Err(err).Msg(err.Error())
+		utils.ErrorWithMessage(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	utils.WithPayload(w, http.StatusOK, res)
+}
+
+func GetTransaction(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+	id := params.ByName("transaction-id")
+
+	transaction, found := models.Transactions[id]
+
+	if !found {
+		msg := "Transaction Not Found"
+		zerolog.TimeFieldFormat = zerolog.TimeFormatUnix
+		log.Info().Msg(msg)
+		utils.ErrorWithMessage(w, http.StatusNotFound, msg)
+		return
+	}
+
+	res, err := jsoniter.Marshal(&transaction)
+	if err != nil {
+		zerolog.TimeFieldFormat = zerolog.TimeFormatUnix
+		log.Error().Err(err).Msg(err.Error())
+		utils.ErrorWithMessage(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	utils.WithPayload(w, http.StatusOK, res)
+}
+
+func PostTransaction(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
```

Comment:
This handler is doing too many things, on the next PR, we will have to create a service layer and split this responsibility into smaller functions

The naming of the function makes me think that it will create a transaction, but internally, it will do
- Deposit
- Withdraw
- Transfer money between accounts

I would consider having a different endpoint for each operation or having on the payload the type of operation that it will do. For example:
```
POST transfer/:from-account-id/to/:to-account-id
OR
POST pay/:account-id HEADER :owner-account-id

POST account/:account-id/deposit
POST account/:account-id/withdraw
```

Reply:
will handle this in a future PR